code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -1,14 +1,28 @@
 /mob/living/carbon/hitby(atom/movable/AM, skipcatch, hitpush = 1, blocked = 0)
 	if(!skipcatch)
-		if(in_throw_mode && !get_active_hand())  //empty active hand and we're in throw mode
+		if(in_throw_mode)  //Makes sure player is in throw mode
```

Comment:
By removing this, you've actually added a bunch of redundant code. Consider that one handed items need the active hand empty, and two handed items need both the active hand and inactive hand empty - you can stack them both behind a requirement for the active hand being empty because they're just layering on top of each other. 

Additionally, by using early returns, you can further reduce the amount of duplicate code, as seen below:
```
if(in_throw_mode && canmove && !restrained())
	if(!istype(AM, /obj/item) || !isturf(AM.loc))
		return FALSE
	if(get_active_hand())
		return FALSE
	if(istype(AM, /obj/item/twohanded))
		if(get_inactive_hand())
			return FALSE
	put_in_active_hand(AM)
	visible_message("<span class='warning'>[src] catches [I]!</span>")
	throw_mode_off()
	return TRUE
```

As for the failure messages, this proc really doesn't need them - it's evident that you didn't catch it by virtue of the fact that it hits you in the face.